### PR TITLE
Add option to draw TouchScreenButton centered, enabled by default

### LINKED ISSUE #22341: TouchScreenButton Rotation behavior
**Godot version:**
3.0.5

**OS/device including version:**
Linux Mint 19 x64

**Issue description:**
Why TouchScreenButton rotates around Position instead of own center (in that time Sprite rotates around center)? Is it possible to rotate TouchScreenButton around center of it's current selected texture?

Thanks.

### PR DESCRIPTION
**Pull request by ignaloidas**
Closes #22341

### CODE REVIEW

File: scene/2d/touch_screen_button.cpp [hunk 1]
```diff
@@ -104,17 +113,29 @@ void TouchScreenButton::_notification(int p_what) {
...
+			}
+
+			if (texture_to_draw.is_valid()) {
+				Point2 offset = Point2();
+				if (draw_centered) {
+					Size2 size = texture_to_draw->get_size();
+					offset -= size / 2;
 				}
+
+				draw_texture(texture, offset);
```
Comment (KoBeWi) [inline scene/2d/touch_screen_button.cpp:138]: Typo.
```suggestion
				draw_texture(texture_to_draw, offset);
```

**Comment by KoBeWi** [comment]
This breaks if texture is centered:
https://github.com/godotengine/godot/blob/b9d877e55fc469e7441cb00856359946b91d00ff/scene/2d/touch_screen_button.cpp#L274-L276
You need to update the condition.

Also why `draw_centered` disables `shape_centered`?

File: scene/2d/touch_screen_button.cpp (hunk 1 above)
Comment (ignaloidas) [reply scene/2d/touch_screen_button.cpp]: Fixed

**Comment by ignaloidas** [comment]
> This breaks if texture is centered:
> 
> https://github.com/godotengine/godot/blob/b9d877e55fc469e7441cb00856359946b91d00ff/scene/2d/touch_screen_button.cpp#L274-L276
> 
> You need to update the condition.

Good catch, fixed.

> Also why `draw_centered` disables `shape_centered`?

`shape_centered` centers the shape in the texture, without it, the shape is centered on `0, 0`, but if `draw_centered` is enabled, the texture is centered on `0, 0` and so `shape_centered` with it's current code would only bring the shape out of the center, and I don't think it makes sense to do anything else when it is disabled.

File: doc/classes/TouchScreenButton.xml [hunk 2]
```diff
@@ -39,7 +42,7 @@
 			The button's shape.
 		</member>
 		<member name="shape_centered" type="bool" setter="set_shape_centered" getter="is_shape_centered" default="true">
-			If [code]true[/code], the button's shape is centered in the provided texture. If no texture is used, this property has no effect.
+			If [code]true[/code], the button's shape is centered in the provided texture. If no texture is used, or [code]shape_centered[/code] is true, this property has no effect.
```
Comment (KoBeWi) [inline doc/classes/TouchScreenButton.xml:45]:
```suggestion
			If [code]true[/code], the button's shape is centered in the provided texture. If no texture is used, or [member shape_centered] is [code]true[/code], this property has no effect.
```

**Review by KoBeWi — APPROVED** [review]
Looks ok functionality and implementation-wise, but the feature itself might still need to be discussed. To me it's a bit weird to have two very similar properties (`draw_centered` and `shape_centered`).

**Comment by mhilbrunner** [comment]
PR meeting: while useful functionality, we're looking to add touch support to the standard button classes instead and remove TouchButton. In those, the feature this PR adds should already be possible with the existing control centering options.

Thanks for contributing!

**Comment by Calinou** [comment]
Would this PR still be relevant for `3.x` where TouchScreenButton will be kept?

**Comment by ignaloidas** [comment]
It breaks compatibility right now by making `draw_centered=true` a default, but if that was changed it is still relevant.

**Pull request closed without merging**